libcaliptra/src/caliptra_api.c

```diff
@@ -1060,4 +1102,53 @@ int caliptra_capabilities(struct caliptra_capabilities_resp *resp, bool async)
     CREATE_PARCEL(p, OP_CAPABILITIES, &checksum, resp);
 
     return pack_and_execute_command(&p, async);
-}
\ No newline at end of file
+}
+
+int caliptra_retrieve_idevid_csr(struct caliptra_buffer* caliptra_idevid_csr) 
+{
+    if (!caliptra_is_idevid_csr_ready()) {
```

Comment:
can we do a param check for caliptra_idevid_csr?

Reply:
done